src/services/api/lib.ts

```diff
@@ -1,44 +1,72 @@
+import { UserData, UserRole } from "../../../types";
 import axios from "./axios";
 import endpoints from "./endpoints";
 
 export type CreateUserProp = {
   username: string;
-  permissions?: string;
+  permissions: UserRole;
   email: string;
 };
 
-type UserData = {
-  data: {
-    permissions: "reviewer" | "admin";
-    id: number;
-    email: string;
-    githubUsername: string;
-    updatedAt: string;
-    createdAt: string;
-    authToken?: string | null;
-    jwt?: string | null;
-    archivedAt?: string | null;
-  };
+type AuthToken = {
+  jwt: string;
 };
 
 export const createNewUser = async ({
   username,
   permissions,
   email,
-}: CreateUserProp): Promise<UserData> => {
+}: CreateUserProp): Promise<{ data: UserData }> => {
   return axios.post(endpoints.USERS(), {
     username,
     permissions,
     email,
   });
 };
 
+export const signUpNewUser = async ({
+  username,
+  permissions,
+  email,
+}: CreateUserProp): Promise<AuthToken & UserData> => {
+  return axios
+    .post(endpoints.USER_SIGN_UP(), {
```

Comment:
You need to pass the github token in the headers as `x-github-token` for new users. That's what the backend uses to create the jwt that is returned

Reply:
good catch